tsdb/batcher.go

```diff
@@ -55,14 +55,23 @@ func (b *PointBatcher) Start() {
 		return
 	}
 
-	var timer *time.Timer
+	// initialize the timer variable
+	timer := &time.NewTimer(b.duration)
```

Comment:
This should just be `timer := time.NewTimer(time.Hour)`, otherwise `timer`'s type is going to be `**time.Timer`. Also, picking an arbitrarily large value is fine, just in case there isn't a duration set, and we're stopping it on the next line, anyway.

Reply:
This doesn't even compile... :-)